R/g_boxplot.R

```diff
@@ -263,7 +263,7 @@ g_boxplot <- function(data,
 
   # Add facetting.
   if (!is.null(facet_var)) {
-    if (facet_var != "None" & facet_var %in% names(data)) {
+    if (facet_var != "None" && facet_var %in% names(data)) {
```

Comment:
I think the original `&` was appropriate.

Reply:
good catch
this indeed change the logic which is out of the scope of this PR
In general I have nothing against fixing bugs as we spot them but this should be noted in PR and re-tested during PR so reviewer needs to be aware of this.

Reply:
I think I remember this as one of the requirements of the current R CMD CHECK (to use `&&` and `||`) in R 4.3.0 but can not reproduce it on the `main` branch when triggering tests


`&&` seems superior to `&` and does less operations in `LHS && RHS` if `LHS` is FALSE. 
if `LHS` is TRUE it does the same as `&` , but when `LHS` is FALSE it stops and does not check any further 

let me revert this change for now

Reply:
yes, but I checked `facet_var` was length one in this case so it's the same for && and & for objects of length 1. anyway I reverted this as I don't see that in R CMD CHECK notes anymore

Reply:
Upon second thought I think my initial comment was wrong. Sorry for creating confusion.

On a slightly unrelated note, I am not sure why `facet_var` has to be checked for NULL and then for "None". If the argument defaults to NULL and then no faceting is applied, why add special handling for "None"?

Reply:
This actually brings this error on LINTR checks - https://github.com/insightsengineering/goshawk/blob/222_decouple_scda%40main/R/g_boxplot.R#L264
I have added #nolint in other PR 34194d2d9ec564fb51c20470df7da709195fe34c to omit that